### PR TITLE
Add play recommendations for song/artist on pressing 'r'

### PR DESCRIPTION
**Pull request by neriglissar**
I implemented a feature which on keypress 'r' loads a list of recommended songs for the selected song or artist and starts playback. This functionality was requested in issue #171 (see also #129). 

Screenshot 2020-01-04 at 09 21 39

### CODE REVIEW

**Comment by Rigellute** [comment]
@neriglissar sorry for the delay in reviewing this. Currently about half way through. 

This looks like an awesome feature!

File: src/app.rs [hunk 1]
```diff
@@ -479,6 +495,61 @@ impl App {
...
+        if let (Some(spotify), Some(user)) = (&self.spotify, &self.user.to_owned()) {
+            let user_country =
+                Country::from_str(&user.country.to_owned().unwrap_or_else(|| "".to_string()));
+            let empty_payload: Map<String, Value> = Map::new();
+
+            match spotify.recommendations(
+                seed_artists, // artists
+                None,         // genres
+                seed_tracks,  // tracks
+                30,           // limit : 20-100 according to reference, but much higher than 30
```
[the remark this answers is not in the document]
Comment (neriglissar) [reply src/app.rs]: Thanks for reviewing! And good suggestion, fixed it now.

**Comment by Rigellute** [comment]
@all-contributors please add @neriglissar for code

**Comment by Rigellute** [comment]
@all-contributors please add @neriglissar for code

**Pull request merged**